moviepy/Clip.py

```diff
@@ -490,10 +497,103 @@ def close(self):
         #    * Therefore, should NOT be called by __del__().
         pass
 
-    # Support the Context Manager protocol, to ensure that resources are cleaned up.
+    # helper private methods
+    def __unsupported(self, other, operator):
+        self_type = type(self).__name__
+        other_type = type(other).__name__
+        message = "unsupported operand type(s) for {}: '{}' and '{}'"
+        raise TypeError(message.format(operator, self_type, other_type))
```

Comment:
It is a little hard to tell, but I _think_ that the pythonic thing to do here is to return `NotImplemented`. I might be wrong, so feel free to disagree.